src/Sentry/SentryStructuredLogger.cs

```diff
@@ -0,0 +1,237 @@
+using Sentry.Extensibility;
+using Sentry.Infrastructure;
+using Sentry.Internal;
+using Sentry.Protocol.Envelopes;
+
+namespace Sentry;
+
+/// <summary>
+/// Creates and sends logs to Sentry.
+/// <para>This API is experimental and it may change in the future.</para>
+/// </summary>
+[Experimental(DiagnosticId.ExperimentalFeature)]
+public sealed class SentryStructuredLogger
+{
+    private readonly IHub _hub;
+    private readonly ISystemClock _clock;
+
+    private readonly SentryOptions? _options;
+    private readonly IInternalScopeManager? _scopeManager;
+
+    internal static SentryStructuredLogger CreateDisabled(IHub hub)
+    {
+        return new SentryStructuredLogger(hub, null, null, SystemClock.Clock);
+    }
+
+    internal SentryStructuredLogger(IHub hub, IInternalScopeManager? scopeManager, SentryOptions? options, ISystemClock clock)
+    {
+        _hub = hub;
+        _clock = clock;
+
+        _options = options;
+        _scopeManager = scopeManager;
+        IsEnabled = options is { EnableLogs: true };
+    }
+
+    [MemberNotNullWhen(true, nameof(_options))]
+    private bool IsEnabled { get; }
+
+    /// <summary>
+    /// Creates and sends a structured log to Sentry, with severity <see cref="SentryLogLevel.Trace"/>, when enabled and sampled.
+    /// <para>This API is experimental and it may change in the future.</para>
+    /// </summary>
+    /// <param name="template">A formattable <see langword="string"/>. When incompatible with the <paramref name="parameters"/>, the log will not be captured. See <see href="https://learn.microsoft.com/dotnet/api/system.string.format">System.String.Format</see>.</param>
+    /// <param name="parameters">The arguments to the <paramref name="template"/>. See <see href="https://learn.microsoft.com/dotnet/api/system.string.format">System.String.Format</see>.</param>
+    /// <param name="configureLog">A delegate to set attributes on the <see cref="SentryLog"/>. When the delegate throws an <see cref="Exception"/> during invocation, the log will not be captured.</param>
+    [Experimental(DiagnosticId.ExperimentalFeature)]
+    public void LogTrace(string template, object[]? parameters = null, Action<SentryLog>? configureLog = null)
+    {
+        if (IsEnabled)
```

Comment:
It looks like all of the public methods have this `IsEnabled` check and then delegate all the work to the `CaptureLog` method.

We could centralise this logic by putting the check for `IsEnabled` at the top of the `CaptureLog` method instead.

Reply:
Alternatively, if we did add an `ISentryStructuredLogger` interface, we could have two implementations of that interface: `SentryStructuredLogger` and `DisabledSentryStructuredLogger`... the implementation for the later would just be a bunch of no-ops. 

One advantage of that approach is that we could mark the dependencies for this implementation as non nullable. I'm not sure if that would simplify things enough to be worth while though (it's a pretty simple class in the first place). 

Your call I think.

Reply:
Changed the `SentryStructuredLogger` to be `abstract`, and centralized all the checks to a single check in the derived `DefaultSentryStructuredLogger` override.